components/function-controller/internal/git/cloner.go

```diff
@@ -0,0 +1,27 @@
+package git
+
+import (
+	git2go "github.com/libgit2/git2go/v31"
+	"github.com/pkg/errors"
+)
+
+type git2goCloner struct {
+}
+
+func (g *git2goCloner) cloneRepo(options Options, outputPath string) (*git2go.Repository, error) {
```

Comment:
What do you think about extracting the whole logic (`GetAuth` method and all `if`'s)? In this case, we'll not test this file anymore and this structure will be still functional